CMakeLists.txt

```diff
@@ -77,6 +78,30 @@ option(
   OFF)
 option(VELOX_MONO_LIBRARY "Build single unified library." OFF)
 option(ENABLE_ALL_WARNINGS "Enable -Wall and -Wextra compiler warnings." ON)
+option(VELOX_BUILD_SHARED "Build Velox as shared libraries." OFF)
+# While it's possible to build both in one go we currently want to build either
+# static or shared.
+cmake_dependent_option(
+  VELOX_BUILD_STATIC
+  "Build Velox as static libraries."
+  ON
+  "NOT VELOX_BUILD_SHARED"
+  OFF)
+
+if(VELOX_BUILD_SHARED AND NOT VELOX_MONO_LIBRARY)
```

Comment:
nit: If we only have VELOX_BUILD_SHARED then we should set VELOX_MONO_LIBRARY to ON if its unset , rather than throw.